third_party/move/move-vm/metrics/src/lib.rs

```diff
@@ -0,0 +1,37 @@
+// Copyright (c) The Move Contributors
+// SPDX-License-Identifier: Apache-2.0
+
+use once_cell::sync::Lazy;
+use prometheus::{register_histogram_vec, HistogramTimer, HistogramVec};
+
+pub trait Timer {
+    fn timer_with_label(&self, label: &str) -> HistogramTimer;
+}
+
+impl Timer for HistogramVec {
+    fn timer_with_label(&self, label: &str) -> HistogramTimer {
+        self.with_label_values(&[label]).start_timer()
+    }
+}
+
+pub static VM_TIMER: Lazy<HistogramVec> = Lazy::new(|| {
```

Comment:
Was not clear to me where the results are dumped/can be accessed? Can we see some results showing all this in action for some transactions, demonstrating that we are getting sane results?

Reply:
The results can be seen on grafana dashboard, I will add a comment for this👍

Reply:
Ah, nice. Can we also see results locally, e.g., when locally replaying a transaction?

Reply:
Yes, if you set up some env variables you can push metrics into grafana, and view them real-time:
```
export PUSH_METRICS_NAMESPACE=...
export PUSH_METRICS_ENDPOINT="..."
```

Reply:
Added a comment about this